src/main/java/spoon/support/StandardEnvironment.java

```diff
@@ -353,13 +353,22 @@ public void setInputClassLoader(ClassLoader aClassLoader) {
 		if (aClassLoader instanceof URLClassLoader) {
 			final URL[] urls = ((URLClassLoader) aClassLoader).getURLs();
 			if (urls != null && urls.length > 0) {
-				List<String> classpath = new ArrayList<>();
+				// Check that the URLs are only file URLs
+				boolean onlyFileURLs = true;
 				for (URL url : urls) {
-					classpath.add(url.toString());
+					if( ! url.getProtocol().equals("file")) {
+						onlyFileURLs = false;
+					}
 				}
-				setSourceClasspath(classpath.toArray(new String[0]));
+				if( onlyFileURLs ) {
```

Comment:
I'm not sure about this check: basically you consider that if your `URLClassLoader` contains other things than local files, you ignore everything silently. 
So:
   1. Why don't you do the loop above here, and at least add the file anyway if it's local, even if the classloader contains other things
   2. To be exhaustive, if URLs do not concern local files, maybe we should write them locally and add their local path, WDYT?

Reply:
> If the URLClassLoader contains anything that is not an URL that refers to a file, we revert to the second alternative (simply setting the class loader in this.classloader).

OK I see your point, but you kept a `return` line 370, meaning, if it's a URLClassloader with other things than paths, the classloader is not set at all... 
Maybe it should be good to add another test to check the behaviour with a URLClassloader with other things than simple paths: it would be our first contract and then we'll be able to evolve it in other PR in the future.